src/Core/src/Platform/iOS/MauiSearchBar.cs

```diff
@@ -51,32 +52,18 @@ protected internal MauiSearchBar(NativeHandle handle) : base(handle)
 			}
 		}
 
+		[UnconditionalSuppressMessage("Memory", "MA0001", Justification = "Proven safe in test: SearchBarTests.DoesNotLeak")]
 		internal event EventHandler? OnMovedToWindow;
-		internal event EventHandler? EditingChanged;
 
 		public override void WillMoveToWindow(UIWindow? window)
 		{
-			var editor = this.GetSearchTextField();
-
 			base.WillMoveToWindow(window);
 
-			if (editor != null)
+			if (this.GetSearchTextField() is UITextField editor)
 				ResignFirstResponderTouchGestureRecognizer.Update(editor, window);
 
-			if (editor != null)
-			{
-				editor.EditingChanged -= OnEditingChanged;
-				if (window != null)
-					editor.EditingChanged += OnEditingChanged;
-			}
```

Comment:
I removed this `EditingChanged` event and subscribe in `MauiSearchBarProxy` instead.